File: tests/java/EmptyCatchClauseTest/EmptyCatchClauseTest.java
// The class has try-catch statement and the catch is empty
// -> Has empty catch smell.
public class EmptyCatchClauseTest //Type originates from: EmptyCatchClauseTest.h
{
	private int f;
	private void func1(int p) //Method definition originates from: EmptyCatchClauseTest.cpp
	{
			try
			{
				if (this.f > p)
				{
					this.f++;
				}
				else
				{
					throw (p);
				}
			}
			catch (int e)
			{

			}
	}
}

File: tests/cpp/EmptyCatchClauseTest/EmptyCatchClauseTest.cpp
// The class has try-catch statement and the catch is empty
// -> Has empty catch smell.
#include "EmptyCatchClauseTest.h"

void EmptyCatchClauseTest :: func1(int p) {
        try {
            if (this->f >p){
                this->f ++;
            } else{
                throw (p);
            }
        }
        catch (int e){

        }
}